src/pages/Appointments/AppointmentsPage.tsx

```diff
@@ -300,8 +294,7 @@ export default function AppointmentsPage(props: { facilityId?: string }) {
 
     // Only update if there are changes
     if (Object.keys(updates).length > 0) {
-      setQParams({
-        ...qParams,
+      updateQuery({
```

Comment:
You can modify the useEffect accordingly as well (removing qParams from dependency for example).